apps/omg_eth/lib/omg_eth/release_tasks/set_contract.ex

```diff
@@ -72,71 +72,76 @@ defmodule OMG.Eth.ReleaseTasks.SetContract do
       end
 
     # get all the data from external sources
-    {payment_exit_game, eth_vault, erc20_vault, min_exit_period_seconds, contract_semver, child_block_interval} =
+    {exit_games, eth_vault, erc20_vault, min_exit_period_seconds, contract_semver, child_block_interval} =
       get_external_data(plasma_framework, rpc_api)
 
+    # Okay, so I am not sure if we want to keep payment_exit_game under contract_addr.
+    # However, this is more backward competible with existing code. If we want to move away
+    # for all the places we should refactor this.
     contract_addresses = %{
       plasma_framework: plasma_framework,
       eth_vault: eth_vault,
       erc20_vault: erc20_vault,
-      payment_exit_game: payment_exit_game
+      payment_exit_game: exit_games.tx_payment_v1,
+      payment_v2_exit_game: exit_games.tx_payment_v2
     }
 
-    merge_configuration(
-      config,
-      txhash_contract,
-      authority_address,
-      contract_addresses,
-      min_exit_period_seconds,
-      contract_semver,
-      network,
-      child_block_interval
-    )
+    extra_config = %{
+      txhash_contract: txhash_contract,
+      authority_address: authority_address,
+      contract_addresses: contract_addresses,
+      exit_games: exit_games,
+      min_exit_period_seconds: min_exit_period_seconds,
+      contract_semver: contract_semver,
+      network: network,
+      child_block_interval: child_block_interval
+    }
+
+    {:ok, []} = valid_extra_config?(extra_config)
+    merge_configuration(config, extra_config)
   end
 
   defp get_external_data(plasma_framework, rpc_api) do
     min_exit_period_seconds = get_min_exit_period(plasma_framework, rpc_api)
 
-    payment_exit_game =
-      plasma_framework |> exit_game_contract_address(ExPlasma.payment_v1(), rpc_api) |> Encoding.to_hex()
+    # TODO: get the list of types from ex_plasma?
+    exit_games =
+      Enum.into(OMG.WireFormatTypes.exit_game_tx_types(), %{}, fn type ->
```

Comment:
Not sure if this version of ex_plasma has these but in childchain v2 I call `ExPlasma.payment_v1()`